modules/nf-core/variancepartition/dream/templates/dream.R

```diff
@@ -147,13 +153,28 @@ fitmm <- eBayes(fitmm, proportion = opt\$proportion,
 head(fitmm\$design, 3)
 print(colnames(fitmm\$design))
 
-# Write coefficient name from contrast variable and target level (e.g., "treatmentmCherry")
-coef_name <- paste0(opt\$contrast_variable, opt\$contrast_target)
+# If contrast_string is provided, use that for makeContrast
+if (!is.null(opt\$contrast_string) && opt\$contrast_string != "") {
+    cat("Using contrast string:", opt\$contrast_string, "\n")
+    # Use makeContrasts
+    contrast_matrix <- makeContrasts(contrasts = opt\$contrast_string, levels = fitmm\$design)
+    fit2 <- contrasts.fit(fitmm, contrast_matrix)
+    fit2 <- eBayes(fit2, proportion = opt\$proportion,
+                   stdev.coef.lim = stdev_coef_lim_vals,
+                   trend = opt\$trend, robust = opt\$robust,
+                   winsor.tail.p = winsor_tail_p_vals)
+    results <- topTable(fit2, number = nrow(countMatrix),
+                        adjust.method = opt\$adjust.method,
+                        p.value = opt\$p.value, lfc = opt\$lfc, confint = opt\$confint)
 
-# Get topTable results
-results <- topTable(fitmm, coef = coef_name,
-                    adjust.method = opt\$adjust.method, p.value = opt\$p.value,
-                    lfc = opt\$lfc, confint = opt\$confint)
+} else {
+    coef_name <- paste0(opt\$contrast_variable, opt\$contrast_target)
+    cat("Using default contrast matrix:", coef_name, "\n")
+
+    results <- topTable(fitmm, coef = coef_name,
+                        adjust.method = opt\$adjust.method, p.value = opt\$p.value,
+                        lfc = opt\$lfc, confint = opt\$confint)
```

Comment:
Is there a reason why you specify `number` explicitly in one case but not the other?